

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Tutorial &mdash; CAPITAL 1.0.0 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="API" href="../api/index.html" />
    <link rel="prev" title="&lt;no title&gt;" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> CAPITAL
          

          
          </a>

          
            
            
              <div class="version">
                1.0.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../Installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Preprocessing-scRNA-seq-data">Preprocessing scRNA-seq data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Computing-a-trajectory-tree-with-CAPITAL">Computing a trajectory tree with CAPITAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Aligning-trajectory-trees-with-CAPITAL">Aligning trajectory trees with CAPITAL</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Aligning-cells-along-each-trajectory-path">Aligning cells along each trajectory path</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Writing-CAPITAL-results">Writing CAPITAL results</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Calculating-similarity-scores-of-genes-in-cell-cell-alignment">Calculating similarity scores of genes in cell-cell alignment</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../api/index.html">API</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">CAPITAL</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="index.html">&lt;no title&gt;</a> &raquo;</li>
        
      <li>Tutorial</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/tutorials/capital_tutorial.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container,
div.nbinput.container div.prompt,
div.nbinput.container div.input_area,
div.nbinput.container div[class*=highlight],
div.nbinput.container div[class*=highlight] pre,
div.nboutput.container,
div.nboutput.container div.prompt,
div.nboutput.container div.output_area,
div.nboutput.container div[class*=highlight],
div.nboutput.container div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt a.copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Tutorial">
<h1>Tutorial<a class="headerlink" href="#Tutorial" title="Permalink to this headline">¶</a></h1>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">capital</span> <span class="k">as</span> <span class="nn">cp</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
</pre></div>
</div>
</div>
<div class="section" id="Preprocessing-scRNA-seq-data">
<h2>Preprocessing scRNA-seq data<a class="headerlink" href="#Preprocessing-scRNA-seq-data" title="Permalink to this headline">¶</a></h2>
<p>Read a count matrix (cells x genes) of your own data as an AnnData object using sc.read().</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1">#adata1 = sc.read(&quot;./your_data1.h5ad&quot;)</span>
</pre></div>
</div>
</div>
<p>Run a simple preproccesing recipe.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1">#cp.tl.preprocessing(adata1, n_Top_genes=2000, K=40)</span>
</pre></div>
</div>
</div>
<p>Specifically, cp.tl.preprocessing() runs a simple preprocessing recipe as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_cells</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_genes</span><span class="o">=</span><span class="n">Min_Genes</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">filter_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_cells</span><span class="o">=</span><span class="n">Min_Cells</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">exclude_highly_expressed</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">highly_variable_genes</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">min_mean</span><span class="o">=</span><span class="n">Min_Mean</span><span class="p">,</span> <span class="n">max_mean</span><span class="o">=</span><span class="n">Max_Mean</span><span class="p">,</span> <span class="n">min_disp</span><span class="o">=</span><span class="n">Min_Disp</span><span class="p">,</span> <span class="n">n_top_genes</span><span class="o">=</span><span class="n">n_Top_genes</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">raw</span> <span class="o">=</span> <span class="n">adata</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">adata</span><span class="p">[:,</span><span class="n">adata</span><span class="o">.</span><span class="n">var</span><span class="p">[</span><span class="s1">&#39;highly_variable&#39;</span><span class="p">]]</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">pca</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">n_comps</span><span class="o">=</span><span class="n">N_pcs</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">neighbors</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">n_neighbors</span><span class="o">=</span><span class="n">K</span><span class="p">,</span> <span class="n">n_pcs</span><span class="o">=</span><span class="n">N_pcs</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">diffmap</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">umap</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">leiden</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">paga</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">groups</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="line-block">
<div class="line">In this tutorial, we will use two datasets that have already been preprocessed.</div>
<div class="line">Download might take a minute.</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">adata1</span> <span class="o">=</span> <span class="n">cp</span><span class="o">.</span><span class="n">dataset</span><span class="o">.</span><span class="n">setty19</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Downloading the dataset.
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
setty19_capital.h5ad: 351MB [02:47, 2.09MB/s]
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Download completed. The dataset is saved in ./capital_dataset/setty19_capital.h5ad
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">adata1</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
AnnData object with n_obs × n_vars = 5780 × 1999
    obs: &#39;n_genes&#39;, &#39;leiden&#39;
    var: &#39;n_cells&#39;, &#39;highly_variable&#39;, &#39;means&#39;, &#39;dispersions&#39;, &#39;dispersions_norm&#39;
    uns: &#39;diffmap_evals&#39;, &#39;hvg&#39;, &#39;leiden&#39;, &#39;leiden_colors&#39;, &#39;leiden_sizes&#39;, &#39;neighbors&#39;, &#39;paga&#39;, &#39;pca&#39;, &#39;umap&#39;
    obsm: &#39;X_diffmap&#39;, &#39;X_pca&#39;, &#39;X_umap&#39;
    varm: &#39;PCs&#39;
    obsp: &#39;connectivities&#39;, &#39;distances&#39;
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">umap</span><span class="p">(</span><span class="n">adata1</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;leiden&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_10_0.png" src="../_images/tutorials_capital_tutorial_10_0.png" />
</div>
</div>
<p>Note that dataset 2 will be loaded later.</p>
</div>
<div class="section" id="Computing-a-trajectory-tree-with-CAPITAL">
<h2>Computing a trajectory tree with CAPITAL<a class="headerlink" href="#Computing-a-trajectory-tree-with-CAPITAL" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line">Pass the AnnData object and the root name in the trajectory to cp.tl.trajectory_tree().</div>
<div class="line">You can pass your metadata in adata.obs by choosing a key in adata.obs.</div>
<div class="line">You can also draw your own defined tree in networkx.DiGraph and pass it to the argument.</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">trajectory_tree</span><span class="p">(</span><span class="n">adata1</span><span class="p">,</span> <span class="n">root_node</span><span class="o">=</span><span class="s2">&quot;4&quot;</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="s2">&quot;leiden&quot;</span><span class="p">,</span> <span class="n">tree</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Draw the calculated tree.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">trajectory_tree</span><span class="p">(</span><span class="n">adata1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_15_0.png" src="../_images/tutorials_capital_tutorial_15_0.png" />
</div>
</div>
<p>Apply the same process to one or more datasets that you would like to align.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">adata2</span> <span class="o">=</span> <span class="n">cp</span><span class="o">.</span><span class="n">dataset</span><span class="o">.</span><span class="n">velten17</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
velten17_capital.h5ad: 0.00B [00:00, ?B/s]
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Downloading the dataset.
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
velten17_capital.h5ad: 41.9MB [00:12, 3.37MB/s]
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Download completed. The dataset is saved in ./capital_dataset/velten17_capital.h5ad
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">adata2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
AnnData object with n_obs × n_vars = 915 × 249
    obs: &#39;n_genes&#39;, &#39;n_counts&#39;, &#39;leiden&#39;
    var: &#39;n_cells&#39;, &#39;highly_variable&#39;, &#39;means&#39;, &#39;dispersions&#39;, &#39;dispersions_norm&#39;
    uns: &#39;diffmap_evals&#39;, &#39;hvg&#39;, &#39;leiden&#39;, &#39;leiden_colors&#39;, &#39;leiden_sizes&#39;, &#39;neighbors&#39;, &#39;paga&#39;, &#39;pca&#39;, &#39;umap&#39;
    obsm: &#39;X_diffmap&#39;, &#39;X_pca&#39;, &#39;X_umap&#39;
    varm: &#39;PCs&#39;
    obsp: &#39;connectivities&#39;, &#39;distances&#39;
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">umap</span><span class="p">(</span><span class="n">adata2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;leiden&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_19_0.png" src="../_images/tutorials_capital_tutorial_19_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">trajectory_tree</span><span class="p">(</span><span class="n">adata2</span><span class="p">,</span> <span class="n">root_node</span><span class="o">=</span><span class="s2">&quot;0&quot;</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="s2">&quot;leiden&quot;</span><span class="p">,</span> <span class="n">tree</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">trajectory_tree</span><span class="p">(</span><span class="n">adata2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_21_0.png" src="../_images/tutorials_capital_tutorial_21_0.png" />
</div>
</div>
</div>
<div class="section" id="Aligning-trajectory-trees-with-CAPITAL">
<h2>Aligning trajectory trees with CAPITAL<a class="headerlink" href="#Aligning-trajectory-trees-with-CAPITAL" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line">The alignment of trajectory trees is calculated by minimizing an alignment distance described in <a class="reference external" href="https://www.biorxiv.org/content/10.1101/859751v1">our paper</a>.</div>
<div class="line">You can specify the number of highly variable genes for both of the data so that the intersection of those genes is used to calculate the tree alignment.</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cdata</span> <span class="o">=</span> <span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">tree_alignment</span><span class="p">(</span><span class="n">adata1</span><span class="p">,</span> <span class="n">adata2</span><span class="p">,</span> <span class="n">num_genes1</span><span class="o">=</span><span class="mi">2000</span><span class="p">,</span> <span class="n">num_genes2</span><span class="o">=</span><span class="mi">2000</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Calculating tree alignment
411 genes are used to calculate cost of tree alignment.

Calculation finished.
</pre></div></div>
</div>
<p>Draw the tree alignment result.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">tree_alignment</span><span class="p">(</span><span class="n">cdata</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_25_0.png" src="../_images/tutorials_capital_tutorial_25_0.png" />
</div>
</div>
<div class="line-block">
<div class="line">Each linear alignment from the root to each leaf (path) is labeled as alignmentXYZ where X, Y and Z indicate digit.</div>
<div class="line">See the list of the alignments name by alignmentlist.</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cdata</span><span class="o">.</span><span class="n">alignmentlist</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[(&#39;alignment000&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;#&#39;, &#39;13&#39;, &#39;1&#39;, &#39;5&#39;, &#39;22&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;17&#39;, &#39;12&#39;, &#39;19&#39;, &#39;18&#39;, &#39;13&#39;]),
 (&#39;alignment001&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;#&#39;, &#39;13&#39;, &#39;1&#39;, &#39;5&#39;, &#39;#&#39;, &#39;11&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;17&#39;, &#39;12&#39;, &#39;19&#39;, &#39;18&#39;, &#39;6&#39;, &#39;9&#39;]),
 (&#39;alignment002&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;14&#39;, &#39;19&#39;, &#39;20&#39;, &#39;2&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;2&#39;, &#39;20&#39;, &#39;#&#39;, &#39;15&#39;]),
 (&#39;alignment003&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;3&#39;, &#39;10&#39;, &#39;12&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;8&#39;, &#39;#&#39;, &#39;#&#39;]),
 (&#39;alignment004&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;3&#39;, &#39;21&#39;, &#39;9&#39;, &#39;15&#39;, &#39;8&#39;, &#39;#&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;8&#39;, &#39;5&#39;, &#39;10&#39;, &#39;21&#39;, &#39;7&#39;, &#39;16&#39;]),
 (&#39;alignment005&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;3&#39;, &#39;21&#39;, &#39;9&#39;, &#39;15&#39;, &#39;8&#39;, &#39;0&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;8&#39;, &#39;5&#39;, &#39;10&#39;, &#39;21&#39;, &#39;7&#39;, &#39;3&#39;]),
 (&#39;alignment006&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;3&#39;, &#39;21&#39;, &#39;9&#39;, &#39;15&#39;, &#39;16&#39;, &#39;18&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;8&#39;, &#39;5&#39;, &#39;10&#39;, &#39;21&#39;, &#39;#&#39;, &#39;11&#39;]),
 (&#39;alignment007&#39;,
  [&#39;4&#39;, &#39;7&#39;, &#39;6&#39;, &#39;3&#39;, &#39;21&#39;, &#39;9&#39;, &#39;15&#39;, &#39;17&#39;],
  [&#39;0&#39;, &#39;1&#39;, &#39;14&#39;, &#39;8&#39;, &#39;5&#39;, &#39;10&#39;, &#39;21&#39;, &#39;4&#39;])]
</pre></div></div>
</div>
</div>
<div class="section" id="Aligning-cells-along-each-trajectory-path">
<h2>Aligning cells along each trajectory path<a class="headerlink" href="#Aligning-cells-along-each-trajectory-path" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line">Run dpt (diffusion pseudotime) to calculate pseudotime for each alignment.</div>
<div class="line">Specify the alignment name (list of alignment names) to choose which path should be compared to calculate cell-cell matching.</div>
<div class="line">Note: calculating pseudotime for all the alignments might take for a minute.</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">dpt</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;alignment001&quot;</span><span class="p">,</span> <span class="s2">&quot;alignment002&quot;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<p>Choose genes to do dynamic time warping.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">genes</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;GATA1&quot;</span><span class="p">,</span> <span class="s2">&quot;CA1&quot;</span><span class="p">,</span> <span class="s2">&quot;LMNA&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">dtw</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment001&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="line-block">
<div class="line">Draw the result of dynamic time warping.</div>
<div class="line">The colors in the labels correspond to the colors of sc.pl.umap().</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">dtw</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment001&quot;</span><span class="p">,</span>
          <span class="n">data1_name</span><span class="o">=</span><span class="s2">&quot;Setty+2019&quot;</span><span class="p">,</span> <span class="n">data2_name</span><span class="o">=</span><span class="s2">&quot;Velten+2017&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_34_0.png" src="../_images/tutorials_capital_tutorial_34_0.png" />
</div>
</div>
<div class="line-block">
<div class="line">Draw gene expression dynamics for both data.</div>
<div class="line">Pseudotime in this figure is matched to adata1’s pseudotime.</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">gene_expression_trend</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment001&quot;</span><span class="p">,</span>
                            <span class="n">data1_name</span><span class="o">=</span><span class="s2">&quot;Setty+2019&quot;</span><span class="p">,</span> <span class="n">data2_name</span><span class="o">=</span><span class="s2">&quot;Velten+2017&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_36_0.png" src="../_images/tutorials_capital_tutorial_36_0.png" />
</div>
</div>
</div>
<div class="section" id="Writing-CAPITAL-results">
<h2>Writing CAPITAL results<a class="headerlink" href="#Writing-CAPITAL-results" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line">Specify a directory so that two AnnData objects, adata1 and adata2, and capital_data.npz are saved in the directory.</div>
<div class="line">You can read CAPITAL data by passing the directory path to cp.tl.read_capital_data().</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[22]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cdata</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;./cdata&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[23]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cdata</span> <span class="o">=</span> <span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">read_capital_data</span><span class="p">(</span><span class="s2">&quot;./cdata&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Calculating-similarity-scores-of-genes-in-cell-cell-alignment">
<h2>Calculating similarity scores of genes in cell-cell alignment<a class="headerlink" href="#Calculating-similarity-scores-of-genes-in-cell-cell-alignment" title="Permalink to this headline">¶</a></h2>
<div class="line-block">
<div class="line">Computing dynamic time warping for each gene will generate a similarity score (see <a class="reference external" href="https://tslearn.readthedocs.io/en/stable/gen_modules/metrics/tslearn.metrics.dtw_path.html#tslearn.metrics.dtw_path">here</a> for details).</div>
<div class="line">You can pass a value to min_disp to choose how many genes are used to calculate the score.</div>
<div class="line">The genes that have larger normalized dispersions per gene than min_disp in either data are chosen.</div>
<div class="line">Results are in cdata.similarity_score.</div>
<div class="line">The higher in the resulting list, the more similar the gene expression trend is.</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[24]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">genes_similarity_score</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment002&quot;</span><span class="p">,</span> <span class="n">min_disp</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Calculating similarity score of 5148 genes in alignment002
Calculating finished
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cdata</span><span class="o">.</span><span class="n">similarity_score</span><span class="p">[</span><span class="s2">&quot;alignment002&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[25]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
array([&#39;TROAP&#39;, &#39;DBI&#39;, &#39;BLOC1S1&#39;, ..., &#39;ARPP21&#39;, &#39;FCRL1&#39;, &#39;NBEAL1&#39;],
      dtype=object)
</pre></div></div>
</div>
<p>See gene expression trends that have similar trends in the alignment.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[26]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">high_similarlity_genes</span> <span class="o">=</span> <span class="n">cdata</span><span class="o">.</span><span class="n">similarity_score</span><span class="p">[</span><span class="s2">&quot;alignment002&quot;</span><span class="p">][:</span><span class="mi">4</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">dtw</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">high_similarlity_genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment002&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[28]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">gene_expression_trend</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">high_similarlity_genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment002&quot;</span><span class="p">,</span>
                            <span class="n">data1_name</span><span class="o">=</span><span class="s2">&quot;Setty+2019&quot;</span><span class="p">,</span> <span class="n">data2_name</span><span class="o">=</span><span class="s2">&quot;Velten+2017&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_46_0.png" src="../_images/tutorials_capital_tutorial_46_0.png" />
</div>
</div>
<p>See also gene expression trends that have lower similarlity score in the alignment.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[29]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">low_similarlity_genes</span> <span class="o">=</span> <span class="n">cdata</span><span class="o">.</span><span class="n">similarity_score</span><span class="p">[</span><span class="s2">&quot;alignment002&quot;</span><span class="p">][::</span><span class="o">-</span><span class="mi">1</span><span class="p">][:</span><span class="mi">4</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">dtw</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">low_similarlity_genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment002&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">cp</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">gene_expression_trend</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">low_similarlity_genes</span><span class="p">,</span> <span class="n">alignment</span><span class="o">=</span><span class="s2">&quot;alignment002&quot;</span><span class="p">,</span>
                            <span class="n">data1_name</span><span class="o">=</span><span class="s2">&quot;Setty+2019&quot;</span><span class="p">,</span> <span class="n">data2_name</span><span class="o">=</span><span class="s2">&quot;Velten+2017&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/tutorials_capital_tutorial_50_0.png" src="../_images/tutorials_capital_tutorial_50_0.png" />
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../api/index.html" class="btn btn-neutral float-right" title="API" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="&lt;no title&gt;" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2021, Reiichi Sugihara, Yuki Kato, Tomoya Mori, Yukio Kawahara.

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>